crates/oxc_linter/src/rules/typescript/prefer_as_const.rs

```diff
@@ -0,0 +1,196 @@
+                        if string_literal.value.to_string().eq(&initial_string.value.to_string()) {
+                            Some(string_literal.span)
+                        } else {
+                            None
+                        }
+                    }
+                    _ => None,
+                },
+                TSLiteral::NullLiteral(null_literal) => match initial_value_expression {
+                    Expression::NullLiteral(_) => Some(null_literal.span),
+                    _ => None,
+                },
+                TSLiteral::NumberLiteral(number_literal) => match initial_value_expression {
+                    Expression::NumberLiteral(initial_number) => {
+                        if number_literal.value == initial_number.value {
+                            Some(number_literal.span)
+                        } else {
+                            None
+                        }
+                    }
+                    _ => None,
+                },
+                _ => None,
+            };
+            match error_span {
+                Some(span) => {
+                    if can_fix {
+                        ctx.diagnostic_with_fix(PreferAsConstDiagnostic(span), || {
+                            let start = span.start;
+                            let end = span.end;
+                            Fix::new("const", Span { start, end })
+                        })
+                    } else {
+                        ctx.diagnostic(PreferAsConstDiagnostic(span));
+                    }
+                }
+                None => {}
+            }
+        }
+        _ => {}
+    }
+}
+
+#[test]
+fn test() {
+    use crate::tester::Tester;
+
+    let pass = vec![
+        "let foo = 'baz' as const;",
+        "let foo = 1 as const;",
+        "let foo = { bar: 'baz' as const };",
+        "let foo = { bar: 1 as const };",
+        "let foo = { bar: 'baz' };",
+        "let foo = { bar: 2 };",
+        "let foo = 'bar' as string;",
+        "let foo = `bar` as `bar`;",
+        "let foo = `bar` as `foo`;",
+        "let foo = `bar` as 'bar';",
+        "let foo: string = 'bar';",
+        "let foo: number = 1;",
+        "let foo: 'bar' = baz;",
+        "let foo = 'bar';",
+        "let foo: 'bar';",
+        "let foo = { bar };",
+        "let foo: 'baz' = 'baz' as const;",
+        "class foo { bar = 'baz'; }",
+        "class foo { bar: 'baz'; }",
+        "class foo { bar; }",
+        "class foo { bar: string = 'baz'; }",
+        "class foo { bar: number = 1; }",
+        "class foo { bar = 'baz' as const; }",
+        "class foo { bar = 2 as const; }",
+        "class foo { get bar(): 'bar' {} set bar(bar: 'bar') {} }",
+        "class foo { bar = () => 'bar' as const; }",
+        "type BazFunction = () => 'baz'; class foo { bar: BazFunction = () => 'bar'; }",
+        "class foo { bar(): void {} }",
+        // NOTE: OXC does not parse these format yet.
```

Comment:
Is my understanding correct?

Reply:
https://www.typescriptlang.org/docs/handbook/jsx.html#the-as-operator

We are parsing everything as tsx, you can keep them commented out.